mods/player_api/api.lua

```diff
@@ -17,80 +14,90 @@ function player_api.register_model(name, def)
 end
 
 -- Player stats and animations
-local player_model = {}
-local player_textures = {}
-local player_anim = {}
-local player_sneak = {}
+-- model, textures, animation
+local players = {}
 player_api.player_attached = {}
 
 function player_api.get_animation(player)
-	local name = player:get_player_name()
-	return {
-		model = player_model[name],
-		textures = player_textures[name],
-		animation = player_anim[name],
-	}
+	return players[player:get_player_name()]
 end
 
 -- Called when a player's appearance needs to be updated
 function player_api.set_model(player, model_name)
-	local name = player:get_player_name()
+	local player_data = players[player:get_player_name()]
+	if player_data.model == model_name then
+		return
+	end
 	local model = models[model_name]
 	if model then
-		if player_model[name] == model_name then
-			return
-		end
-		player:set_properties({
+		player:set_properties{
 			mesh = model_name,
-			textures = player_textures[name] or model.textures,
+			textures = player_data.textures or model.textures,
 			visual = "mesh",
 			visual_size = model.visual_size or {x = 1, y = 1},
 			collisionbox = model.collisionbox or {-0.3, 0.0, -0.3, 0.3, 1.7, 0.3},
 			stepheight = model.stepheight or 0.6,
 			eye_height = model.eye_height or 1.47,
-		})
+		}
+		local animations = model.animations
+		player:set_local_animation(
+			animations.stand,
+			animations.walk,
+			animations.mine,
+			animations.walk_mine,
+			model.animation_speed or 30
+		)
 		player_api.set_animation(player, "stand")
 	else
-		player:set_properties({
+		player:set_properties{
 			textures = {"player.png", "player_back.png"},
 			visual = "upright_sprite",
 			visual_size = {x = 1, y = 2},
 			collisionbox = {-0.3, 0.0, -0.3, 0.3, 1.75, 0.3},
 			stepheight = 0.6,
 			eye_height = 1.625,
-		})
+		}
 	end
-	player_model[name] = model_name
+	player_data.model = model_name
 end
 
 function player_api.set_textures(player, textures)
-	local name = player:get_player_name()
-	local model = models[player_model[name]]
-	local model_textures = model and model.textures or nil
-	player_textures[name] = textures or model_textures
-	player:set_properties({textures = textures or model_textures})
+	local player_data = players[player:get_player_name()]
+	local model = models[player_data.model]
+	local new_textures = model and model.textures or textures
+	player_data.textures = new_textures
+	player:set_properties{textures = new_textures}
```

Comment:
I don't know if our code style requires this, but it's done like this everywhere else.
```suggestion
	player:set_properties({textures = new_textures})
```

Reply:
I prefer the shorter notation. Can change though.

Reply:
Whereas this is a technically correct notation, it is not used in Minetest's code. New modders might take this code as a reference, and will be confused by this awkward special case allowed by Lua.